### PR TITLE
test: end to end test coding

### PR DESCRIPTION
**Pull request by ramshty**
#### **Description:**
This PR introduces a comprehensive end-to-end (E2E) test for the `MainActivity` of the QuickFix app. The test covers the full user flow from the welcome screen to the registration screen, ensuring that the UI elements are correctly displayed, interactable, and that the navigation works as expected. All test cases pass successfully, and the feature is ready for review.

#### **Key Changes:**
- Added a new test class `MainActivityTest` using **Kaspresso** and **Compose UI Test**.
- The test verifies:
  - Proper rendering and interaction with the welcome screen (`WelcomeScreen`).
  - Navigation from the welcome screen to the registration screen upon clicking the "Register" button.
  - Input fields on the `RegisterScreen` (e.g., first name, birth date) are visible, focusable, and accept user input.
  - Correct handling of edge cases such as delayed UI rendering by utilizing `waitUntil` conditions where necessary.
  - Ensured the test performs with retry logic for stability in case of UI lags or flakiness.

#### **Highlights:**
- All test cases pass without issues, confirming that the app's UI flow from the welcome screen to the registration screen is working as intended.
- Logging was added to track interactions and help in debugging any future issues.
- The implementation also ensures no performance bottlenecks, with frame rendering being stable throughout the test run.

#### **Test Status:**
- The test has been run successfully on multiple devices and configurations.
- No issues were encountered during testing, and all tests are passing.

#### **Next Steps:**
- Review the PR and provide feedback.
- Merge the PR upon approval to ensure robust test coverage for the `MainActivity`.

### CODE REVIEW

File: app/src/androidTest/java/com/arygm/quickfix/kaspresso/MainActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      }
+      composeTestRule.mainClock.advanceTimeBy(1500L)
+      composeTestRule.onNodeWithTag("inputEmail").performTextInput("main.activity@test.com")
+      composeTestRule.onNodeWithTag("inputPassword").performTextInput("246890357Asefthuk")
+      composeTestRule.onNodeWithTag("logInButton").assertIsEnabled()
+      composeTestRule.onNodeWithTag("logInButton").performClick()
+      composeTestRule.onRoot().printToLog("TAG")
+    }
+  }
+}
```
Comment (JitterKiller) [inline app/src/androidTest/java/com/arygm/quickfix/kaspresso/MainActivityTest.kt:83]: We could also test the bottom bar/navigating screen to screen but great start!

**Review by JitterKiller — APPROVED** [review]
Excellent work on this end-to-end test! The use of Kaspresso is impressive. Good Job!

Maybe for next e2e tests, add additional tests on the bottom bar e.g navigating screen to screen.

LGTM!
